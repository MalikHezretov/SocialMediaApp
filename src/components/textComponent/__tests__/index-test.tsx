import 'react-native';
import React from 'react';
import TextComponent from '..';
import renderer from 'react-test-renderer';

it('renders TextComponent correctly', () => {
  const testRenderer = renderer.create(
    <TextComponent title="test" description="description" />,
  );
  const testInstance = testRenderer.root;

  expect(testInstance.findByType(TextComponent).props.title).toBe('test');
  expect(testInstance.findByType(TextComponent).props.description).toBe(
    'description',
  );
  expect(testRenderer).toMatchSnapshot();
});

it('renders TextComponent with empty description', () => {
  const testRenderer = renderer.create(<TextComponent description="" />);
  const testInstance = testRenderer.root;

  expect(testInstance.findByType(TextComponent).props.description).toBe('');
  expect(testRenderer).toMatchSnapshot();
});
