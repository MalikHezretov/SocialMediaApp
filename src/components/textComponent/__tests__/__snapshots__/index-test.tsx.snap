// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextComponent correctly 1`] = `
<View
  style={
    Object {
      "padding": 16,
      "paddingBottom": 0,
    }
  }
>
  <Text>
    test
  </Text>
  <Text
    numberOfLines={1}
  >
    description
  </Text>
</View>
`;

exports[`renders TextComponent with empty description 1`] = `
<View
  style={
    Object {
      "padding": 16,
      "paddingBottom": 0,
    }
  }
>
  <Text>
    Title
  </Text>
  <Text
    numberOfLines={1}
  >
    
  </Text>
</View>
`;
